Sources/Components/Feed/InAppFeedViewModel.swift

```diff
@@ -0,0 +1,312 @@
+            do {
+                let _ = try await Knock.shared.messageModule.updateMessageStatus(messageId: item.id, status: .unread)
+                
+                if let index = feed.entries.firstIndex(where: { $0.id == item.id }) {
+                    await MainActor.run {
+                        feed.entries[index].read_at = nil
+                    }
+                }
+                
+                await fetchNewMetaData()
+            } catch {
+                Knock.shared.log(type: .error, category: .feed, message: "error in markAsUnread: \(error.localizedDescription)")
+            }
+        }
+        
+        public func markAsInteracted(_ item: Knock.FeedItem) async {
+            do {
+                let _ = try await Knock.shared.messageModule.updateMessageStatus(messageId: item.id, status: .interacted)
+            } catch {
+                Knock.shared.log(type: .error, category: .feed, message: "error in markAsInteracted: \(error.localizedDescription)")
+            }
+        }
+        
+        // MARK: FeedItemRow Interactions
+        public func feedItemButtonTapped(item: Knock.FeedItem, actionString: String) {
+            didTapFeedItemButtonPublisher.send(actionString)
+        }
+        
+        public func feedItemRowTapped(item: Knock.FeedItem) {
+            didTapFeedItemRowPublisher.send(item)
+            Task {
+                await markAsInteracted(item)
+            }
+        }
+        
+        // MARK: Button/Swipe Interactions
+        
+        /// Called when a user performs a horizontal swipe action on a row item.
+        public func didSwipeRow(item: Knock.FeedItem, swipeAction: FeedNotificationRowSwipeAction) {
+            Task {
+                switch swipeAction {
+                case .archive: await archiveItem(item)
+                case .markAsRead: await markAsRead(item)
+                case .markAsUnread: await markAsUnread(item)
+                }
+            }
+        }
+        
+        /// Called when a user taps on one of the action buttons at the top of the list.
+        public func topActionButtonTapped(action: Knock.FeedTopActionButtonType) async {
+            switch action {
+            case .archiveAll(_):
+                await archiveAll(scope: .all)
+            case .archiveRead(_):
+                await archiveAll(scope: .read)
+            case .markAllAsRead(_):
+                await markAllAsRead()
+            }
+        }
+        
+        // MARK: Private Methods
+        
+        private func filterDidChange() {
+            self.feedClientOptions.status = self.currentFilter.scope
+
+            Task { [weak self] in
+                await self?.refreshFeed(showRefreshIndicator: true)
+            }
+        }
+        
+        private func fetchNewMetaData() async {
+            do {
+                let options = Knock.FeedClientOptions(tenant: currentTenantId, has_tenant: feedClientOptions.has_tenant)
```

Comment:
Should we not be passing all of the options that are set here? E.g. if I applied `archived` filter we'd be returning the wrong metadata here